cmd/fluent-bit/loki.go

```diff
@@ -226,12 +227,17 @@ func createLine(records map[string]interface{}, f format) (string, error) {
 				return "", nil
 			}
 		}
-		return buf.String(), nil
+		return replaceNewLineAndTab(buf.String()), nil
 	default:
 		return "", fmt.Errorf("invalid line format: %v", f)
 	}
 }
 
+func replaceNewLineAndTab(line string) string{
+	replacer := strings.NewReplacer(`\n`, "\n", `\t`, "\t")
+	return replacer.Replace(line)
+}
+
```

Comment:
This is a hacky solution. Somewhere the encoding by Json and logfmt is treating `\n` and `\t` as escaped. So, we have to replace those with actual `\n` and `\t`.

Any better ways?

Reply:
well I guess the replacer can be global for sure.

Reply:
And so you probably don't need this function.

Reply:
Yeah looks like a fine solution to me. Basically unescaping those characters.

Reply:
Have you tried to see what fluentd send for those ? I'm curious to see if they do the same.

Reply:
Fluent-bit records are fine as well. I just checked the content before Marshaling the record here 
https://github.com/grafana/loki/blob/7399beff4aaacb1a1c0c925101d88bedf2cd9666/cmd/fluent-bit/loki.go#L203
`\n` and `\t` looks fine here. 

But after the marshaling, result from `jsoniter.ConfigCompatibleWithStandardLibrary.Marshal(records)` doesn't respect `\n` and `\t`. Looks like Marshaling is escaping these characters. 

Same applies to `logfmt` encoding as well
https://github.com/grafana/loki/blob/7399beff4aaacb1a1c0c925101d88bedf2cd9666/cmd/fluent-bit/loki.go#L217

Reply:
Which is correct everything is enclosed by "" and \ must be escaped. May be this is more a Grafana issue ? when reading \\n it should read \n ? cc @aocenas WDYT ?

Reply:
Yeah, this can be handled from Grafana side as well. However, line would be this `Hello\nworld` and not `Hello\\nworld`. So, on Grafana they can replace `\n` with actual newline.